src/process.ts

```diff
@@ -128,12 +128,18 @@ export class PowerShellProcess {
         this.consoleCloseSubscription = vscode.window.onDidCloseTerminal((terminal) => this.onTerminalClose(terminal));
 
         // Log that the PowerShell terminal process has been started
-        const pid = await this.consoleTerminal.processId;
+        const pid = await this.getPid();
         this.logTerminalPid(pid ?? 0, pwshName);
 
         return sessionDetails;
     }
 
+    // Returns the process Id of the consoleTerminal
+    public async getPid() {
```

Comment:
```suggestion
    public async getPid(): number | undefined {
```

I'm surprised ESLint isn't complaining about untyped functions, I would want that auto-enforced...

Reply:
Oh gosh, it's:

```
    "@typescript-eslint/explicit-function-return-type": [
      "error"
    ],
```

and enabling it currently results in 96 problems...my bad! That's a tech debt day item to fix for sure.

Reply:
To be fair the functions aren't untyped per se, they're inferred from the return type and all the type checking flows from that (and the type is visible in the inlay hints). I personally prefer that but I get if you want to enforce that style.